[PicukiBridge] fix images not displaying

User123698745 opened the pull request:
Images are currently not displayed (It used to work some weeks ago), because the image urls contain special characters like + or |, which can not be properly handled by all rss readers.

This PR ensures that post image gets included in the feed items:
- The last path segment gets encoded (main fix)
- `#.jpg` gets added to the url path, so that it get handles as `image/jpeg` instead of `application/octet-stream` (Note: Only `.jpg` without `#` does not work, because it breaks the link)
- The image gets included in the item content, increasing the chance for it being displayed, instead of relying on rss viewers to handle enclosures

Minor additional changes:
- The `(video)` note is now italic to make it clear that it is not part of the post description
- Cutoff titles now end with `...` to make it clear that it is not the full post description
- The username gets added as author

User123698745 committed:
[PicukiBridge] fix image not displaying

Bockiii commented:
Well, it doesnt work in the html preview. Can you show me the difference in your reader? (screenshots of the current and your version)

User123698745 commented:
Right sorry, I forgot to mention that the rss bridge preview does not work.

You can use my test heroku instance: https://11179f88d728.herokuapp.com

Current: https://bridge.easter.fr/?action=display&bridge=Picuki&format=Atom&context=Username&u=aesoprockwins
Changed: https://11179f88d728.herokuapp.com/?action=display&bridge=Picuki&format=Atom&context=Username&u=aesoprockwins

Livemarks (Firefox Extension: https://addons.mozilla.org/en-US/firefox/addon/livemarks):
| Current | Changed |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/1911682/167687866-81f60d4c-aea6-48b6-879a-670e40c4026e.png) | ![image](https://user-images.githubusercontent.com/1911682/167687911-b1abb5a5-5ecc-450a-bb08-3ffcde12c59e.png) |

Inoreader (https://www.inoreader.com):
| Current | Changed |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/1911682/167686915-05dfd703-9084-4fbd-8756-9f6e1b55bf9f.png) | ![image](https://user-images.githubusercontent.com/1911682/167687018-607dc2f1-f3b1-4f40-bdff-426681bbdbfc.png) |

Thunderbird:
| Current | Changed |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/1911682/167686533-67faddeb-0553-4d6a-858f-997e5eb3a18b.png) |  ![image](https://user-images.githubusercontent.com/1911682/167686571-7796325a-00f3-45c1-ba9f-df38ed2acd0e.png) |

Note: While making the screenshots I noticed I forgot to mention an additional change, that the username now gets added as author (I now added it to the description)

Bockiii commented:
Thanks, great work, I like it.
